typescript/sdk/src/deploy/HyperlaneAppChecker.ts

```diff
@@ -106,15 +97,36 @@ export abstract class HyperlaneAppChecker<
             provider,
             contract.address,
           );
-          if (!eqAddress(actualProxyAdminAddress, expectedProxyAdminAddress)) {
+          const actualProxyAdminContract = ProxyAdmin__factory.connect(
+            actualProxyAdminAddress,
+            provider,
+          );
+          const actualProxyAdminOwner = await actualProxyAdminContract.owner();
+          const expectedOwner = ownableOverrides?.['proxyAdmin'] ?? owner;
+          if (!eqAddress(actualProxyAdminOwner, expectedOwner)) {
```

Comment:
do we expect / want this code to run if `expectedProxyAdminAddress` is defined? if not, can we explicitly skip this?

I think we want to make sure that we don't ever create two OwnerViolations. If we call it once here and again later, I think that'd be a bug

Reply:
I suppose it is safe to assume that that if `expectedProxyAdminAddress` is defined that the ownership check will be done in `checkOwnership` so we can skip it here